### PR TITLE
S4 Header Footer

### PR DESCRIPTION
**Pull request by kuzivka**

### CODE REVIEW

File: src/components/footer/FoodTags.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function FoodTags() {
+  return (
+    <>
+      <Typography sx={footerTagsTitle} variant="h6">
+        Product tags
+      </Typography>
+      <Box sx={footerTagsContainer}>
+        {food.map((name, i) => (
+          <Chip key={i} label={name} />
```
Comment (Kost927) [inline src/components/footer/FoodTags.tsx]: use index for the key is not good. You can use useId() hook from react 18, or .name if the name is unique

File: src/components/footer/FooterLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  footerLinksLink,
+  footerLinksTitle,
+} from '@styles/footer-styles';
+import { links } from './constants';
+
+export default function FooterLinks() {
+  return (
+    <Box sx={footerLinksContainer}>
+      {links.map(({ title, links }, index) => (
+        <Box key={index}>
```
Comment (Kost927) [inline src/components/footer/FooterLinks.tsx]: key={index}, also somewhere you have "i", somewhere "index".  Should be one code style

File: src/components/footer/FooterLinks.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          <Typography component="div" variant="h6" sx={footerLinksTitle}>
+            {title}
+          </Typography>
+          <List>
+            {links.map(({ link, resource }, i) => (
```
Comment (Kost927) [inline src/components/footer/FooterLinks.tsx]: same

File: src/components/header/CategoryMenus.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { headerCategoryMenu } from '@styles/header-styles';
+import DropDownMenu from './DropDownMenu';
+
+export default function CategoryMenu() {
+  const { data: categories } = useGetCategoriesQuery('');
+
+  return (
+    <List sx={headerCategoryMenu}>
+      {categories &&
+        categories.map((category, i) => <DropDownMenu key={i} {...category} />)}
```
Comment (Kost927) [inline src/components/header/CategoryMenus.tsx]: .

File: src/components/header/DropDownMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <ListItem>
+      <Button
+        sx={headerMenuButton}
+        id="fade-button"
+        aria-haspopup="true"
+        aria-expanded={open ? 'true' : undefined}
```
Comment (Kost927) [inline src/components/header/DropDownMenu.tsx]: logic inside JSX

File: src/components/header/DropDownMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        TransitionComponent={Fade}
+        open={open}
+        MenuListProps={{
+          onMouseLeave: handleClose,
+          sx: categoryMenuOptions,
+        }}
+        onClose={handleClose}
+      >
+        {brand.map(({ name }, i) => (
+          <MenuItem key={i} sx={categoryMenuOptions} onClick={handleClose}>
```
Comment (Kost927) [inline src/components/header/DropDownMenu.tsx]: .

File: src/components/header/SearchInput.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  const [selectState, setSelectState] = useState('All Categories');
+  const { data: categories } = useGetCategoriesQuery('');
+  return (
+    <Paper elevation={0} component="form" sx={searchInput}>
+      <Select
+        disableUnderline
+        variant="standard"
+        value={selectState}
+        labelId="demo-simple-select-label"
+        id="demo-simple-select"
```
Comment (Kost927) [inline src/components/header/SearchInput.tsx]: id, labelId are not related to your project

File: src/components/header/SearchInput.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        label="All"
+        onChange={(e) => {
+          setSelectState(e.target.value);
+        }}
+        IconComponent={SelectArrowIcon}
+        sx={searchInputSelect}
+      >
+        <MenuItem value={'All Categories'}>All Categories</MenuItem>
+        {categories &&
+          categories.map(({ name }, i) => (
```
Comment (Kost927) [inline src/components/header/SearchInput.tsx]: .

File: src/components/header/SearchInput.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  searchInputDivider,
+  searchInputField,
+  searchInputSelect,
+} from '@styles/header-styles';
+import { useState } from 'react';
+import { SelectArrowIcon } from './SelectArrowIcon';
+
+export default function SearchInput() {
+  const [selectState, setSelectState] = useState('All Categories');
+  const { data: categories } = useGetCategoriesQuery('');
```
Comment (Kost927) [inline src/components/header/SearchInput.tsx]: Screenshot 2022-10-07 at 15 32 55
You don't pass any arguments, so you do not need to pass an empty string to your useGetCategoriesQuery()

File: src/components/footer/FoodTags.tsx (hunk 1 above)
Comment (kuzivka) [reply src/components/footer/FoodTags.tsx]: about the useId() hook I found this (it says that it's not a good practice to use it for keys):
https://twitter.com/reactjs/status/1510334505252491267?s=20&t=OXCRI-sGD6YwBJFcVrd-Ew

**Review by Kost927 — APPROVED** [review]

**Pull request merged**